### PR TITLE
Change route to ssl, use annotations in Golang template

### PR DESCRIPTION
**Pull request by PhilipGough**
@wtrocki @maleck13

### CODE REVIEW

File: haproxy-config.template [hunk 1]
```diff
@@ -144,9 +144,11 @@ backend be_edge_http_{{$cfgIdx}}
   cookie {{$cfg.RoutingKeyName}} insert indirect nocache httponly
   {{ end }}
   http-request set-header Forwarded for=%[src];host=%[req.hdr(host)];proto=%[req.hdr(X-Forwarded-Proto)]
-  reqirep ^([^\ :]*)\ {{$cfg.Path}}/(.*)  \1\ /\2
-  reqirep ^([^\ :]*)\ {{$cfg.Path}}(.*)  \1\ /\2
-  reqadd X-Script-Name:\ {{$cfg.Path}}
+  {{ with $value := index $cfg.Annotations "subdomain"}}
+  reqirep ^([^\ :]*)\ /{{$value}}/(.*)  \1\ /\2
+  reqirep ^([^\ :]*)\ /{{$value}}(.*)  \1\ /\2
```
Comment (wtrocki) [inline haproxy-config.template]: we can replace this with one reqirep:
`reqirep ^([^\ :]*)\ /{{$value}}/?(.*)  \1\ /\2`

File: haproxy-config.template [hunk 2]
```diff
@@ -239,7 +241,9 @@ backend be_tcp_{{$cfgIdx}}
 {{ define "/var/lib/haproxy/conf/os_edge_http_be.map" }}
 {{ range $idx, $cfg := .State }}
   {{ if and (ne $cfg.Host "") (eq $cfg.TLSTermination "edge")}}
-{{ $cfg.Path }} {{$idx}}
+  {{ with $value := index $cfg.Annotations "subdomain"}}
+/{{$value}} {{$idx}}
```
Comment (wtrocki) [inline haproxy-config.template]: 👍

File: haproxy-config.template (hunk 1 above)
Comment (PhilipGough) [inline haproxy-config.template]: Possibly but then we seem to need to append a / on t the end of the route to hit it

Comment (PhilipGough) [inline haproxy-config.template]: @wtrocki I tested this out and it works for the client app without the trailing slash but not when viewing in the browser. Whats your opinion on removing this?

**Pull request merged**